lib/inspec-reporter-json-hdf/reporter.rb

```diff
@@ -0,0 +1,162 @@
+
+require 'inspec/plugin/v2'
+require 'json'
+
+VALID_FREQUENCY = %w[annually semiannually quarterly monthly every2weeks weekly every3days daily].freeze
+
+VALID_STATUSES = %w[passed failed].freeze
+
+DATE_FORMAT = '%Y-%m-%d'.freeze
+
+module InspecPlugins::HdfReporter
+  # Reporter Plugin Class
+  class Reporter < Inspec.plugin(2, :reporter)
+    def render
+      output(report.to_json, false)
+    end
+
+    def self.run_data_schema_constraints
+      '~> 0.0' # Accept any non-breaking change
+    end
+
+    def report
+      report = Inspec::Reporters::Json.new(@config).report
+      attestations = collect_attestations
+
+      report[:profiles].each do |profile|
+        profile[:controls].each do |control|
+          attestation = attestations.detect { |x| x['control_id'].eql?(control[:id]) }
+
+          next if attestation.nil?
+
+          control[:attestation] = attestation
+          unless attestation_expired?(attestation['updated'], attestation['frequency'])
+            control[:results] = apply_attestation(control[:results], attestation)
+          end
+        end
+      end
+      report
+    end
+
+    private
+
+    def apply_attestation(results, attestation)
+      if results.empty?
+        results = [{
+          "code_desc": 'Manually verified Status provided through attestation',
+          "run_time": 0.0,
+          "start_time": DateTime.now.to_s,
+          "status": attestation['status'],
+          "message": attestation_message(attestation)
+        }]
+      else
+        results.each do |result|
+          result[:message] = 'Automated test returned as passed.' if result[:status].eql?('passed')
+          result[:message] = result[:skip_message] if result[:status].eql?('skipped')
+
+          result[:status] = attestation['status']
+          result[:message] = result[:message] + attestation_message(attestation)
+
+          if result[:backtrace]
+            result[:message] = result[:message] + "\nbacktrace: #{result[:backtrace]}"
+            result[:backtrace] = nil
+          end
+        end
+      end
+      results
+    end
+
+    def attestation_message(attestation)
+      "
+       Attestation:
+       Status: #{attestation['status']}
+       Explanation: #{attestation['explanation']}
+       Updated: #{attestation['updated']}
+       Updated By: #{attestation['updated_by']}
+       Frequency: #{attestation['frequency']}
+       "
+    end
+
+    def attestation_expired?(date, frequency)
+      (advanced_date(date, frequency).mjd - DateTime.now.mjd).negative?
```

Comment:
Since these are both datetime objects can't they just be compared?

```ruby
advanced_date(date, frequency) < DateTime.now
```

Reply:
Duh... thanks